package/nmea_daemon/src/nmea_daemon.c

```diff
@@ -16,64 +16,63 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <czmq.h>
-
-#include <libpiksi/util.h>
 #include <libpiksi/logging.h>
+#include <libpiksi/endpoint.h>
+#include <libpiksi/loop.h>
 
 #define PROGRAM_NAME      "nmea_daemon"
 
 #define NMEA_PUB_ENDPOINT ">tcp://127.0.0.1:44030"  /* NMEA Pub */
 
 #define BASE_DIRECTORY    "/var/run/nmea"
 
-const char*const NMEA_GGA_OUTPUT_PATH = BASE_DIRECTORY "/GGA";
+const char* const NMEA_GGA_OUTPUT_PATH = BASE_DIRECTORY "/GGA";
 
 bool nmea_debug = false;
 
+typedef struct nmea_ctx_s {
+  pk_endpoint_t *sub_ept;
+  pk_loop_t *loop;
+  int result;
+} nmea_ctx_t;
```

Comment:
Was there a benefit of adding this context other than being able to clean-up "cleanly"?  This is much cleaner, but in general Linux will clean-up all "dangling" sockets and open FDs when a process exits.

Reply:
Yea I think I'm overly wary of the side-effects of just exiting anywhere in the program. I realize now that it's probably overkill. I would be fine to leave it in it's original form, this is kind of like the program-level version of 'multiple returns' 😆

Reply:
I definitely prefer the cleaner approach, makes the code much easier to reason about and modularize, but feel free to the "exit hammer" as time saving measure if needed 😃